.github/workflows/tests.yml

```diff
@@ -0,0 +1,202 @@
+name: Tests
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+
+      - name: Run linter
+        run: |
+          tox -e pep8
+
+  unit:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.12"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+
+      - name: Run Unit tests
+        run: |
+          tox -e py3
+
+  coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+
+      - name: Run coverage
+        run: |
+          tox -e cover
+
+  build:
+    name: Build charm
+    needs:
+      - lint
+      - unit
+      - coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+          sudo snap install charmcraft --channel 3.x/stable --classic
+          sudo snap install lxd
```

Comment:
Should we pin this to a specific channel?

Reply:
I wasn't aware of `charmcraft` being sensitive to LXD versions, but I guess there's always a chance of breaking API.  We can pin it to `5.21` (LTS)